docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,694 @@
+        * **subjectAltNames**: list of names to verify in the certificate.
+        * **caCert**: defines a certificate of CA.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientCert**: defines a certificate of a client.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientKey**: defines a client private key.
+          * one of `inline`, `inlineString` or `secret`.
+* **status**: status of an object managed by Kuma control-plane
+  * **vip**: section for allocated IP
+    * **value**: allocated IP for a provided domain with `KumaHostname` type in a match section or provided IP
+    * **type**: provides information about the way IP was provided
+  * **addresses**: section for generated domain
+    * **hostname**: generated domain 
+    * **status**: indicate if an address is available
+    * **origin**: section providing information what generated the vip
+      * **kind**: points to kind that generated domain
+      * **name**: name of the `HostnameGenerator`
+    * **reason**: holds error messages if there are any 
+
+#### Default CA certificate
+
+By default, when TLS is enabled and CA certificate is not set, we are going to use system bundled one. We can achive that by scanning in `kuma-dp` [default paths](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl#example-configuration) for each environment and later send that information in the bootstrap request to the control-plane for a configuration generation.
+
+#### Policy matching
+
+`MeshExternalServices` should work the same with policies as `MeshService`. We are going to introduce a new `kind: MeshExternalService`, which allows targeting them with policies.
+
+#### Envoy resources naming
+
+Currently, we name clusters after their respective policies. Since the model includes only one match we can stick to the same pattern but prefix it with `meshexternalservice_`.
+Example:
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: httpbin
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  type: Managed # Managed
+  managed:
+    match:
+      type: Domain # Kuma will generate a domain
+      value: httpbin.com
+      port: 80
+      protocol: http
+    extension:
+      type: Lambda 
+      config: # type JSON
+        arn: arn:aws:lambda:us-west-2:123456789012:function:my-function
+    endpoints:
+      - address: 1.1.1.1
+        port: 12345
+```
+
+Cluster name: `meshexternalservice_{policyName}`, so for the policy name above, it would look like: `meshexternalservice_httpbin`.
+
+Listener name: we don't need new listeners, only filter chain matches. We will create filter chain matches to match traffic. For TLS and TCP traffic, we will name them `meshexternalservice_{policyName}` because each listener points to the cluster. For `HTTP` traffic, we will name them `meshexternalservice_http_{port}` because we can distinguish the traffic on routes based on host header.
```

Comment:
```suggestion
Listener name: we don't need new listeners, only filter chain matches. We will create filter chain matches to match traffic. For TLS and TCP traffic, we will name them `meshexternalservice_{resourceName}` because each listener points to the cluster. For `HTTP` traffic, we will name them `meshexternalservice_http_{port}` because we can distinguish the traffic on routes based on host header.
```

MES is a **resource** not a **policy**